packages/unmock-core/src/service/service.ts

```diff
@@ -0,0 +1,63 @@
+import { filter as _filter } from "lodash";
+import { HTTPMethod, IStateMapping, OASSchema } from "./interfaces";
+
+/**
+ * Implements the state management for a service
+ */
+
+// TODO: Is there room here for xstate, in the future?
+
+export class Service {
```

Comment:
Should this implement an `IService` interface?

Reply:
Sure - I don't mind that, I don't see the specific need for it at this point. Do you reckon there will be other classes implementing `IService`?

Reply:
No I don't, but I don't think the general principle "program against an interface, not an object" really requires that? But it can be a lone class for now.

Reply:
Fair enough, will add as a `TODO` for the next PR.